packages/@react-stately/grid/src/GridCollection.ts

```diff
@@ -42,7 +42,23 @@ export class GridCollection<T> implements IGridCollection<T> {
 
       let childKeys = new Set();
       let last: GridNode<T> | null = null;
-      for (let child of node.childNodes) {
+      let rowHasCellWithColSpan = false;
+
+      if (node.type === 'item') {
+        for (let child of node.childNodes) {
+          if (child.props?.colSpan !== undefined) {
+            rowHasCellWithColSpan = true;
+            break;
+          }
+        }
+      }
+
+      for (let child of node.childNodes as Iterable<GridNode<T>>) {
+        if (child.type === 'cell' && rowHasCellWithColSpan) {
+          child.colspan = child.props?.colSpan;
+          child.colIndex = !last ? child.index : (last.colIndex ?? last.index) + (last.colspan ?? 1);
+        }
```

Comment:
we have access to column count here, we should do the same check as we're doing in RAC, otherwise this is the message in React Spectrum

`Uncaught TypeError: Cannot read properties of undefined (reading 'props')`

Reply:
Yes, I tried adding it there but got a lot of failing tests before I figured out I needed to check that node is of `item` type 😅 .

I've added it now.